File: BookStore-Angular-MongoDb/angular/src/app/author/models/update-author-dto.ts

export class UpdateAuthorDto  {
  name: string;
  birthDate: string;
  shortBio: string;

  constructor(initialValues: Partial<UpdateAuthorDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}

File: BookStore-Blazor-EfCore/src/Acme.BookStore.Application.Contracts/Authors/UpdateAuthorDto.cs
﻿using System;
using System.ComponentModel.DataAnnotations;

namespace Acme.BookStore.Authors;

public class UpdateAuthorDto
{
    [Required]
    [StringLength(AuthorConsts.MaxNameLength)]
    public string Name { get; set; } = string.Empty;

    [Required]
    public DateTime BirthDate { get; set; }

    public string? ShortBio { get; set; }
}
